### ISSUE TITLE
BuildFields type is incorrect?

### ISSUE THREAD

**Issue by daa** [hasura/ra-data-hasura#91]
``BuildFields`` type is defined as a function returning ``IntrospectionObjectType[]``:
```
type BuildFields = (
  type: IntrospectionObjectType,
  aorFetchType?: FetchType
) => IntrospectionObjectType[];
```

But ``buildFields`` function actually constructs ``FieldNode[]`` list accumulating ``gqlTypes.field()`` results which are ``FieldNode`` objects:
```
const buildFields: BuildFields = (type) =>
  type.fields.reduce((acc, field) => {
    const type = getFinalType(field.type);

    if (type.kind !== TypeKind.OBJECT && type.kind !== TypeKind.INTERFACE) {
      return [...acc, gqlTypes.field(gqlTypes.name(field.name))];
    }

    return acc;
  }, [] as IntrospectionObjectType[]);
```

May be I don't understand something but actual value returned from ``buildFields()`` function look like ``FieldNode[]`` and not like ``IntrospectionObjectType[]`` and they are used where ``FieldNode[]`` is expected and not ``IntrospectionNodeType[]``: https://github.com/hasura/ra-data-hasura/blob/b0f8cabdf2a0a79db7056b049128d0e7c01899de/src/buildGqlQuery/index.ts#L56-L71.

**Comment by cpv123** [comment]
@daa you are correct, good spot. I've raised this PR to fix it: https://github.com/hasura/ra-data-hasura/pull/93

**Comment by daa** [comment]
Thank you very much!

**Comment by cpv123** [comment]
This was fixed in #93
